### PR TITLE
Fix min_by/max_by(x, y, n)

### PR DESCRIPTION
**Pull request by kagamiori**
Summary:
Same as bug in min/max(x, n) fixed in https://github.com/facebookincubator/velox/pull/8311, min_by/max_by(x, y, n) also breaks the 
assumption of incremental window aggregation because their extractValues() methods 
has a side effect of clearing the accumulator.

This diff fixes this issue by making the extractValues() methods of min_by/max_by(x, y, n) 
not clear the accumulators.

Since Presto's min_by/max_by have the same bug (https://github.com/prestodb/presto/issues/21653). This fix 
will make Velox's min_by/max_by behave differently from Presto when used in Window 
operation, until https://github.com/prestodb/presto/issues/21653 is fixed.

This diff fixes https://github.com/facebookincubator/velox/issues/8138.

Differential Revision: D53139892

### CODE REVIEW

File: velox/functions/prestosql/aggregates/MinMaxByAggregates.cpp
Comment (bikramSingh91) [inline velox/functions/prestosql/aggregates/MinMaxByAggregates.cpp:172]: nit: might be worth mentioning with a single line comment that we are returning in ascending order since it looks like we have maintained that expectation here as compared to the previous implementation using PQ. Also, is this a required expectation on the order here?

File: velox/functions/prestosql/aggregates/MinMaxByAggregates.cpp
Comment (bikramSingh91) [inline velox/functions/prestosql/aggregates/MinMaxByAggregates.cpp:254]: nit: maybe we can just call 

```
base.heapValues.pop_back()
addToAccumulator(comparison, value, comparator);
```
We can then get rid of `addToFullAccumulator()`
`pop_back()` does not modify capacity and only modifies the size, so should be equivalent to what you want to achieve in `addToFullAccumulator()`

File: velox/functions/prestosql/aggregates/tests/MinMaxByAggregationTest.cpp
Comment (bikramSingh91) [inline velox/functions/prestosql/aggregates/tests/MinMaxByAggregationTest.cpp:1766]: can you please remind me what this does and why we moved it here

File: velox/functions/prestosql/aggregates/tests/MinMaxByAggregationTest.cpp
Comment (bikramSingh91) [inline velox/functions/prestosql/aggregates/tests/MinMaxByAggregationTest.cpp:2118]: can you add a comment describing what this is testing/verifying?

File: velox/functions/prestosql/aggregates/tests/MinMaxByAggregationTest.cpp
Comment (kagamiori) [reply velox/functions/prestosql/aggregates/tests/MinMaxByAggregationTest.cpp:1766]: testIncremental() is used for testing with valid inputs and testIncremental internally uses global aggregation. The tests under variableN are invalid inputs with global aggregation, so I disable it in this unit test.

File: velox/functions/prestosql/aggregates/MinMaxByAggregates.cpp
Comment (bikramSingh91) [reply velox/functions/prestosql/aggregates/MinMaxByAggregates.cpp]: @kagamiori I think i mistakingly wrote ascending order instead of descending in my comment. Your previous change was correct and my comment was just asking you to write the order in a code comment. The original implementation used PQ and pulling one value at a time from there would return the results in descending order (which is what you were doing earlier by iterating from size()-1 to 0 index of the array).

Secondly, Can you please check if there indeed is any strict order that is expected here?

Comment (bikramSingh91) [reply velox/functions/prestosql/aggregates/MinMaxByAggregates.cpp]: @kagamiori Thanks for the detailed reply, looks like I got confused in my second review instead haha😅

**Review by bikramSingh91 — APPROVED** [review]

**Pull request closed without merging**
